Don't rely on timescaledb.restoring for upgrade

mkindahl opened the pull request:
If a binary upgrade is in progress (when using `pg_upgrade`) the
per-database setting of `timescaledb.restoring` can be included in the
dump, which causes `pg_upgrade` to fail.

This commit fixes this by checking the global variable
`IsBinaryUpgrade` and not refreshing cache if we are in the middle of
doing a binary upgrade.

Fixes #1844

mkindahl commented:
Tested manually in CI.

k-rus commented:
> Tested manually in CI.

Do you have a link?

BTW why no Travis build is run on this PR?

mkindahl commented:
> > Tested manually in CI.
> 
> Do you have a link?

Updated previous message to clarify situation.

> 
> BTW why no Travis build is run on this PR?

No idea. I changed this back and forth between draft status and review status, so that might have caused issues. Anyway, pushed a fix for Windows issue, so now it is started.
